tests/integration/targets/lambda/tasks/main.yml

```diff
@@ -315,49 +315,49 @@
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).policy is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).mappings is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).tags is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).architectures == ['arm64']
 
   - name: lambda_info | Ensure default query value is 'config' when function name
       omitted
     lambda_info:
     register: lambda_infos_query_config
     check_mode: yes
-  - name: lambda_info | Assert successfull retrieval of all information
+  - name: lambda_info | Assert successfull retrieval of all information 2
     assert:
       that:
       - lambda_infos_query_config is not failed
       - lambda_infos_query_config.functions | length > 0
-      - lambda_infos_query_config.functions[0].function_name == lambda_function_name
-      - lambda_infos_query_config.functions[0].runtime == lambda_python_runtime
-      - lambda_infos_query_config.functions[0].description == ""
-      - lambda_infos_query_config.functions[0].function_arn is defined
-      - lambda_infos_query_config.functions[0].handler == lambda_python_handler
-      - lambda_infos_query_config.functions[0].versions is not defined
-      - lambda_infos_query_config.functions[0].aliases is not defined
-      - lambda_infos_query_config.functions[0].policy is not defined
-      - lambda_infos_query_config.functions[0].mappings is not defined
-      - lambda_infos_query_config.functions[0].tags is not defined
+      - lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | length == 1
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).runtime == lambda_python_runtime
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).description == ""
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).function_arn is defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).handler == lambda_python_handler
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).versions is not defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).aliases is not defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).policy is not defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).mappings is not defined
+      - (lambda_infos_all.functions | selectattr('function_name', 'eq', lambda_function_name) | first).tags is not defined
```

Comment:
This block should be testing `lambda_infos_query_config` rather than `lambda_infos_all`

I also find that `in` behaves more consistently than `defined`:  `- '"policy" not in ...'` or `- '"function_arn" in ...'` There are some gotchas around exactly what counts as "defined" when working in Ansible/Jinja2